### PR TITLE
add open store retry

### PR DESCRIPTION
**Pull request by c4pt0r**

### CODE REVIEW

**Comment by c4pt0r** [comment]
@shenli @morefreeze PTAL

File: tidb.go [hunk 1]
```diff
@@ -299,7 +305,17 @@ func NewStore(uri string) (kv.Storage, error) {
...
-	s, err := d.Open(schema)
+	var err error
+	var s kv.Storage
+	for i := 1; i <= maxRetries; i++ {
+		s, err = d.Open(schema)
+		if err == nil || !kv.IsRetryableError(err) {
+			break
+		}
+		sleepTime := time.Duration(uint64(retrySleepInterval) * uint64(i))
+		log.Errorf("Waiting store to get ready, sleep %v and try again...", sleepTime)
```
Comment (morefreeze) [inline tidb.go]: how about log.Warnf

File: tidb_test.go [hunk 2]
```diff
@@ -306,6 +313,15 @@ func (s *testMainSuite) TestitrimSQL(c *C) {
 	}
 }
 
+func (s *testMainSuite) TestRetryOpenStore(c *C) {
+	begin := time.Now()
+	RegisterStore("dummy", &brokenStore{})
+	_, err := newStoreWithRetry("dummy://dummy-store", 3)
+	c.Assert(err, NotNil)
+	elapse := time.Since(begin)
+	c.Assert(uint64(elapse), Greater, uint64(2*time.Second))
```
Comment (morefreeze) [inline tidb_test.go]: 0.5+1+1.5>=3s?

**Comment by shenli** [comment]
LGTM

**Comment by morefreeze** [comment]
LGTM

**Pull request merged**